base_time_window/models/time_weekday.py

```diff
@@ -60,3 +60,20 @@ def unlink(self):
         result = super().unlink()
         self._get_id_by_name.clear_cache(self)
         return result
+
+    def _get_next_weekday_date(self, date_from=False, include_date_from=True):
+        """Returns the next Date matching weekday
+
+        :param date_from: Date object from which we start searching for next weekday
+        :param include_date_from: Allows to return date from if it's same weekday
+        :return Date object matching the weekday
+        """
+        self.ensure_one()
+        if not date_from:
+            date_from = fields.Date.today()
+        next_date = date_from
+        if next_date.weekday() == int(self.name) and not include_date_from:
+            next_date = fields.Date.add(next_date, days=1)
+        while next_date.weekday() != int(self.name):
+            next_date = fields.Date.add(next_date, days=1)
+        return next_date
```

Comment:
```suggestion
        next_date = date_from
        if next_date.weekday() == int(self.name) and not include_date_from:
            next_date = fields.Date.add(next_date, days=1)
        return next_date + timedelta(days=7)
```

Reply:
I think it improves readability by a factor of + infinity ^^

Reply:
````
next_weekday_delta = int(self.name) - date_from.weekday
if next_weekday_delta < 0:
    next_weekday_delta += 7
elif next_weekday_delta == 0 and not_include_date_from:
    next_weekday_delta += 7
return datefrom + timedelta(days=next_weekday_delta)
```

At least this is what I would write from what I understand of the docstring.

Reply:
@gurneyalex Nice to directly compute the delta